src/main/java/com/google/sps/model/ActionableMessageHelperImpl.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.sps.exceptions.GmailMessageFormatException;
+import com.google.sps.utility.GmailUtility;
+import java.util.Arrays;
+import java.util.List;
+
+/** Contains methods that help create the non-trivial fields in an ActionableMessage. */
+public final class ActionableMessageHelperImpl implements ActionableMessageHelper {
+  // Public for testing purposes
+  public static final String UNDISCLOSED_RECIPIENTS_TO_HEADER = "undisclosed-recipients:;";
+  public static final int MAILING_LIST_LOWER_BOUND = 15;
+
+  @Override
+  public ActionableMessage.MessagePriority assignMessagePriority(
+      Message message, String userEmail) {
+    if (isStarred(message)) {
+      return ActionableMessage.MessagePriority.HIGH;
+    }
+
+    if (!isFromMailingList(message, userEmail) || isImportant(message)) {
+      return ActionableMessage.MessagePriority.MEDIUM;
+    }
+
+    return ActionableMessage.MessagePriority.LOW;
+  }
+
+  /**
+   * If "STARRED" is included in a message's labelIds, it has been starred by the user
+   *
+   * @param message Message object to be checked
+   * @return true if starred, false otherwise
+   */
+  private boolean isStarred(Message message) {
+    List<String> labelIds = message.getLabelIds();
+    return labelIds != null && labelIds.contains("STARRED");
+  }
+
+  /**
+   * If the user's email is not in the "To" header, or if there are more than
+   * MAILING_LIST_LOWER_BOUND recipients, the message is considered to be sent from a mailing list.
+   * This is somewhat of a naive approach, however it should still be relatively accurate
+   * (particularly based on the first check listed)
+   *
+   * @param message Message object to be checked
+   * @param userEmail the email address of the current user. Used to check if the email was actually
+   *     sent to them or sent as a part of a mailing list
+   * @return true if from mailing list, false otherwise
+   * @throws GmailMessageFormatException if the message does not contain a "To" header (which is a
+   *     mandatory header, indicating the Message object is not of the correct type [METADATA with
+   *     correct header settings, or FULL])
+   */
+  private boolean isFromMailingList(Message message, String userEmail) {
```

Comment:
I think this method should be named something more general like `isBulkMail` or `isGroupMail`. I think this is a good prioritization criteria, but it's a lot more general than just "sent from a mailing list".

If you want to check for an actual mailing list, there are headers included by most mailing list software that will tell you it's from a mailing list, and even help identify the precedence/priority: https://stackoverflow.com/q/6378773

Reply:
I agree - I'll change up the name

I also added a check looking for a `List-ID` header, which (if present) means the email is likely from a mailing list. Thanks for sharing this!